### ISSUE TITLE
[MU4 Issue] "Height" text edit field for vertical frames shows incorrect value

### ISSUE THREAD

**Issue by bkunda** [musescore/MuseScore#9841]
**Describe the bug**
There are a couple of related parts to this issue:
1. Changes made to the **Height** text edit field for vertical frames do not affect the actual frame height until only _after_ the user has manually adjusted the frame height with the grip handle. 
2. Manual adjustments to the height of a vertical frame only appear in the properties panel after the frame has been deselected and reselected.

https://user-images.githubusercontent.com/86290556/143019796-59786d69-e973-420d-982d-7214bfd114a8.mp4

**Expected behavior**
There should be a real time interface between the **Height** text field in the **Properties** panel and the corresponding vertical frame. 
Moreover, the user should be able adjust the frame height using either the text edit field or the grip handle, and in either case the frame height should be altered accordingly with the correct height value rendered in the text edit field. 

**Desktop (please complete the following information):**
 - MacOS 11.6 (M1)
 - Windows 11

**Comment by Srijan-Chak012** [comment]
Hi! Can I try this issue out?

**Comment by bkunda** [comment]
Please do, @Srijan-Chak012! Many thanks in advance 🙏🏻

**Comment by Srijan-Chak012** [comment]
Hey! the issue is not reproducable in my linux system though. Only the second problem described, the ones where the height from the box can be changed only after clicking the handle, seems to occur in linux. I hae attached a video on how it shows up on linux 

https://user-images.githubusercontent.com/76106446/148190516-87a7f7ba-bbbb-41c6-ba37-d54864b8b88c.mp4

**Comment by bkunda** [comment]
Interesting. I've just tested again in MacOS and it seems that vertical frames inserted as part of a template file (or indeed, as part of any new file) seem to adjust correctly from the Properties panel, but new vertical text frames don't (they require the grip handle to be moved before the text field becomes active). 

![text-frame-resize](https://user-images.githubusercontent.com/86290556/148197775-32cfcebc-4fb0-4ff9-8f5c-4cde0c92259f.gif)

Not sure if this helps you in your detective work, but there's definitely still a problem there. Good luck!

**Comment by SHedge117** [comment]
Hello! I've been looking into this issue a little bit and was wondering if someone has found exactly where in the code this bug is occurring. If not, perhaps someone has an idea on a good place to start to look?

**Comment by bkunda** [comment]
> Hello! I've been looking into this issue a little bit and was wondering if someone has found exactly where in the code this bug is occurring. If not, perhaps someone has an idea on a good place to start to look?

@cbjeukendrup could you please help @SHedge117 with this question?

**Comment by SHedge117** [comment]
@cbjeukendrup Sounds good! I will look into that. Thank you for the help!

**Comment by rgreen5** [comment]
One other thing (if it hasn't already been mentioned.) If you press the frame "Height" Reset button, the height always resets to Zero. AFAIK, this should reset to the default value (10sp?). This causes the frame to disappear from view. If focus is subsequently moved away from the frame it cannot be reselected!

**Comment by wizofaus** [comment]
Is it possible for repro steps to be added to this issue?   I'm not seeing that adjusting the height of a vertical frame via the Properties panel is not updating the score at any rate. I am seeing the reverse (that the properties panel isn't updated after dragging to adjust height).

@bkunda

**Comment by wizofaus** [comment]
FWIW, it seems all the various Inspector "settings" models require explicitly setting up the updatePropertiesOnNotationChanged handler to deal with this, I'm not sure why it can't just be more generic, other than possible efficiency concerns.  I've added one for VerticalFrameSettingsModel and TextLineSettingsModel (and removed the one for HairpinSettingsModel) which handles the cases mentioned above, but there are undoubtedly other similar cases.

**Comment by rgreen5** [comment]
If it hasn't been done already, the readouts for frame gaps and margins need to be standardised to show two decimal places (as in MS3) – starting at 0.00.

**Comment by wizofaus** [comment]
Can you log that as a separate bug then, thanks - not sure how many places it occurs currently, but obviously would be nice to have it handled in one central place for all relevant properties.

**Comment by iainhallam** [comment]
@RomanPudashkin It looks as though the PR for this fixes the updating of the value in properties, but in the released 4.2 I still can't use the properties panel to resize a vertical frame until it's been manually resized (presumably turning off autoSizeEnabled). Should that be separated into a new issue?
